riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -31,19 +36,36 @@ class DeployFailureNotifications(config: Config,
     prefix + path.url
   }
 
+  def getAwsAccountIdTarget(target: ci.Target, parameters: DeployParameters): Option[Target] = {
+    Try {
+      val keyring = lookup.keyRing(parameters.stage, MagentaApp(target.app), MagentaStack(target.stack))
+      STS.withSTSclient(keyring, Region(target.region)){ client =>
+            AwsAccount(STS.getAccountNumber(client))
+      }
+    } match {
+      case Success(value) =>
+        Some(value)
+      case Failure(exception) =>
+        log.error("Failed to fetch AWS account ID", exception)
+        None
+    }
+  }
+
   def failedDeployNotification(uuid: UUID, parameters: DeployParameters) = {
     val deriveAnghammaradTargets = for {
       yaml <- targetResolver.fetchYaml(parameters.build)
       deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
     } yield {
       TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
-        List(App(target.app), Stack(target.stack))
+        List(App(target.app), Stack(target.stack))++ getAwsAccountIdTarget(target, parameters).toList
       } ++ List(Stage(parameters.stage.name))
     }
 
+
     deriveAnghammaradTargets match {
       case Right(targets) =>
-        val failureMessage = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed."
+        log.info(s"Targets to notify: ${targets.toSet}")
+        val failureMessage = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed. Targets: $targets"
```

Comment:
do we still need to the targets variable at the end of this message?

Reply:
Good spot, I've removed it